web/src/pages/api/user/update.ts

```diff
@@ -0,0 +1,121 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import sharp from "sharp";
+
+import { prisma } from "../../../server/db/client";
+import { validateRequest } from "../../../utils/jwt";
+
+interface Request extends NextApiRequest {
+  body: {
+    userId: string;
+    image: string; // base64
+    userName: string;
+  };
+}
+
+export const config = {
+  api: {
+    bodyParser: {
+      sizeLimit: "10mb",
+    },
+  },
+};
+
+export default async function update(req: Request, res: NextApiResponse) {
+  // Validate if the user has a valid JWT token
+  if (!(await validateRequest(req))) {
+    return res.status(401).json({ error: "You are not logged in" });
+  }
+
+  const { userId, image, userName } = req.body;
+
+  if (!image && !userName) {
+    return res.status(400).json({ error: "No image or name provided" });
+  }
+
+  let uploadedImageURL: string | null = null;
+  if (image) {
+    uploadedImageURL = await uploadImage(image, res);
+    if (!uploadedImageURL) {
+      return;
+    }
+  }
+
+  if (userName) {
+    const user = await prisma.user.findUnique({
+      where: {
+        name: userName,
+      },
+    });
+    if (user) {
+      return res.status(400).json({ error: "User name already exists" });
+    }
+  }
+
+  prisma.user
+    .update({
+      where: {
+        id: userId,
+      },
+      data: {
+        image: uploadedImageURL || undefined,
+        name: userName || undefined,
+      },
+    })
+    .then((user) => {
+      if (!user) {
+        // This should never happen but there might be a case where the account is no longer in the database but
+        // the JWT token is still valid
+        return res.status(404).json({ error: "User not found" });
+      }
+
+      res.status(200).json(user);
+    })
+    .catch((err) => {
+      console.error(err);
+      res.status(500).json({ error: "Internal database error" });
+    });
+}
+
+async function uploadImage(image: string, res: NextApiResponse) {
+  // Resize and compress the image before uploading it to imgur
+  // Sharp will resize  both images and animated gifs
+  const uri = image.split(";base64,").pop() as string;
```

Comment:
I would use `.at(-1)` instead of `.pop()` unless you have a good reason I don't know about.